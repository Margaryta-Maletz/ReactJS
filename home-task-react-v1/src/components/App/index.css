.wrapper {
    width: 1200px;
    box-sizing: border-box;
    padding: 20px 60px;
    display: flex;
}
