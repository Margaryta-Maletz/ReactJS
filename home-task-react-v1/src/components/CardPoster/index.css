.card-poster_wrapper {
    width: 322px;
    display: flex;
    flex-wrap: wrap;
    justify-content: space-between;

    font-family: Montserrat;
    font-style: normal;
    font-weight: 500;
    color: white;
    mix-blend-mode: normal;
}

.card-poster_title {
    font-size: 18px;
    line-height: 22px;
    opacity: 0.7;
    margin: 22px 0 8px;
    width: 75%;
}

.card-poster_genre {
    font-size: 14px;
    line-height: 17px;
    opacity: 0.5;
    margin-block: 0;
    margin: 0 0 50px;
}

.card-poster_release-date {
    width: 66px;
    height: 26px;
    border: 1px solid #555555;
    box-sizing: border-box;
    border-radius: 4px;
    font-size: 14px;
    line-height: 24px;
    text-align: center;
    margin: 22px 0 8px;
}
