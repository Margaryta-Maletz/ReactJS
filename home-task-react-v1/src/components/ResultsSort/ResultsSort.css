.results-sort_header {
    font-family: Montserrat;
    font-style: normal;
    font-weight: normal;
    font-size: 16px;
    line-height: 20px;
    letter-spacing: 1px;
    text-transform: uppercase;
    color: darkgrey;
    mix-blend-mode: normal;
}

.results-sort_list {
    margin-left: 31px;
    font-family: Montserrat;
    font-style: normal;
    font-weight: 500;
    font-size: 16px;
    line-height: 20px;
    letter-spacing: 1px;
    text-transform: uppercase;
    color: white;
    background: none;
    border: none;
}

.result-sort-item {
    background: #424242;
}
