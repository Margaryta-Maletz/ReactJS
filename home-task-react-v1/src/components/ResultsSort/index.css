.results-sort_header {
    font-family: Montserrat;
    font-style: normal;
    font-weight: normal;
    font-size: 16px;
    line-height: 20px;
    letter-spacing: 1px;
    text-transform: uppercase;
    color: white;
    mix-blend-mode: normal;
    opacity: 0.6;
}

.results-sort_list {
    margin-left: 31px;
    font-family: Montserrat;
    font-style: normal;
    font-weight: 500;
    font-size: 16px;
    line-height: 20px;
    letter-spacing: 1px;
    text-transform: uppercase;
    color: white;
    background: none;
    border: none;
}
