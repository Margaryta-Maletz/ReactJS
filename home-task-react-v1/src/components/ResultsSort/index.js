import React from 'react';
import './index.css';

export const ResultsSort = (props) => {
    return (
        <>
            <label className='results-sort_header'>
                sort by
                <select className='results-sort_list'>
                    {props.sort.map((item, index) =>
                        index ? <option className='result-sort-item'>{ item }</option>
                            : <option selected className='result-sort-item'>{ item }</option>
                    )}
                </select>
            </label>
        </>
    )
}
