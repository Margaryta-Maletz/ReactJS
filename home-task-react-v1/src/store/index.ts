import thunk from 'redux-thunk';
import {configureStore, getDefaultMiddleware, ThunkAction, Action} from '@reduxjs/toolkit';
import rootReducer from './slice';

const middleware = getDefaultMiddleware({
    imitableCheck: false,
    serializableCheck: false,
    thunk: true,
});

export const store = configureStore({
    reducer: rootReducer,
    middleware: middleware,
    devTools: process.env.NODE_ENV !== 'production',
});

/*export type AppDispatch = typeof store.dispatch;
export type RootState = ReturnType<typeof store.getState>;
export type AppThunk<ReturnType = void> =
    ThunkAction<
        ReturnType,
        RootState,
        unknown,
        Action<string>
    >;*/
