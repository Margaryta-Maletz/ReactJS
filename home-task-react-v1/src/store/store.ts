import {createStore, applyMiddleware} from 'redux';
import thunk from 'redux-thunk';
import {configureStore, ThunkAction, Action} from '@reduxjs/toolkit';
import rootReducer from './reducers';

export const store = createStore(
    rootReducer,
    applyMiddleware(thunk)
);

export type AppDispatch = typeof store.dispatch;
export type RootState = ReturnType<typeof store.getState>;
export type AppThunk<ReturnType = void> =
    ThunkAction<
        ReturnType,
        RootState,
        unknown,
        Action<string>
    >;
