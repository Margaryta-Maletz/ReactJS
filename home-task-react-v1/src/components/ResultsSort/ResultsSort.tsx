import React from 'react';
import './ResultsSort.css';
import {sortList} from '../../consts';
import {useDispatch, useSelector} from "react-redux";
import {IState} from "../../store/types";
import {setSortItem} from "../../store/slice";

export const ResultsSort: React.FC = () => {
    const dispatch = useDispatch();
    const { sortItem } = useSelector((state: IState) => state);

    const handleSortItem = (e: React.ChangeEvent<HTMLSelectElement>) => {
        dispatch(setSortItem(e.target.value));
    }

    return (
        <>
            <label className='results-sort_header'>
                sort by
                <select className='results-sort_list' value={sortItem} onChange={handleSortItem}>
                    {sortList.map((elem) => (
                        <option key={elem.id} value={elem.id} className='result-sort-item'>{elem.value}</option>
                    ))}
                </select>
            </label>
        </>
    )
}
