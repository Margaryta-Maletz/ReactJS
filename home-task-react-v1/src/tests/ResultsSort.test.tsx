import React from "react";
import { render } from '@testing-library/react';
import { ResultsSort } from '../components/ResultsSort/';
import {Provider} from "react-redux";
import { store } from '../store/store'

jest.mock("react-router-dom", () => ({
    ...jest.requireActual("react-router-dom"),
    useLocation: () => ({
        pathname: "localhost:8080/search/"
    }),
    useHistory: (path: string) => ({
        push: jest.fn(),
        location: {
            pathname: "localhost:8080/search/" + path
        }
    })
}));


describe('make snapshot', () => {
    test('renders', () => {
        const { asFragment } = render(
            <Provider store={store}>
                <ResultsSort />
            </Provider>
        );
        expect(asFragment).toMatchSnapshot();
    });
});
