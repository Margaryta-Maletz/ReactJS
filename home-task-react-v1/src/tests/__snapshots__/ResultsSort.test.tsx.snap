// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`make snapshot renders 1`] = `
<div>
  <label
    class="results-sort_header"
  >
    sort by
    <select
      class="results-sort_list"
    >
      <option
        class="result-sort-item"
        value=""
      >
        rating
      </option>
      <option
        class="result-sort-item"
        value="release_date"
      >
        Release Date
      </option>
      <option
        class="result-sort-item"
        value="title"
      >
        Title
      </option>
      <option
        class="result-sort-item"
        value="genre"
      >
        Genre
      </option>
    </select>
  </label>
</div>
`;
